crates/loader/src/local/mod.rs

```diff
@@ -125,26 +130,38 @@ pub fn validate_raw_app_manifest(raw: &RawAppManifestAnyVersion) -> Result<()> {
     Ok(())
 }
 
+fn filter_components(components: &Vec<OsString>, c: &RawComponentManifest) -> bool {
```

Comment:
Maybe just pass the component id here?  It doesn't really need to know the whole manifest.